### PR TITLE
[mark-flow-ui] Use language-aware strings on ballot review page

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

Replacing strings shown on the ballot review pages in Mark/Mark-Scan with language-aware `<UiString>` versions.

Updated the undervote count warning text as described [here](https://docs.google.com/document/d/1xtt5q1Pet5j1RHVsduvOcRGrqDkGspFeKhKc1aVscKE/edit?usp=sharing), in order to avoid interpolated audio.

## Demo Video or Screenshot
|  |  |
| ------------- | ------------- |
|  Screenshot 2023-10-30 at 14 53 14 | Screenshot 2023-10-30 at 15 09 37 |

## Testing Plan
- Manually testing out translated text for now - integration tests and lint checks later down the line
- Existing tests as regression guards

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/ui/src/ui_strings/app_strings.tsx [hunk 1]
```diff
@@ -45,6 +71,31 @@ export const appStrings = {
...
+    <UiString uiStringKey="instructionsBmdReviewPageChangingVotes">
+      To change your vote in any contest, use the select button to navigate to
+      that contest. When you are finished making your ballot selections and
+      ready to print your ballot, use the right button to print your ballot.
+    </UiString>
+  ),
+
+  // TODO(kofi): I think these instructions could be improved a bit. Not sure
+  // it's obvious to a vision-impaired voter that they have to navigate down to
+  // the confirm/reject buttons at the bottom of the screen.
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/app_strings.tsx:91]: Good call!

File: libs/ui/src/ui_strings/app_strings.tsx [hunk 2]
```diff
@@ -59,6 +110,16 @@ export const appStrings = {
...
+    </UiString>
+  ),
+
+  labelWriteInParenthesized: () => (
+    <UiString uiStringKey="labelWriteInParenthesized">(write-in)</UiString>
+  ),
+
   number: (value: number) => <NumberString value={value} />,
 
   numSeatsInstructions: (numSeats: number) =>
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/app_strings.tsx:125]: nit: I see that you've started converging on the pattern of prefixing keys, e.g. `label...`, `instructions...`, `button...`, `title...`, which I like! This one could be renamed `instructionsNumSeats` accordingly

**Review by arsalansufi — APPROVED** [review]
LGTM!

File: libs/ui/src/ui_strings/app_strings.tsx (hunk 2 above)
Comment (kofi-q) [reply libs/ui/src/ui_strings/app_strings.tsx:125]: yeah, good call! will update the older keys to reflect the new pattern

Comment (kofi-q) [reply libs/ui/src/ui_strings/app_strings.tsx:126]: ✅  [7b302e1](https://github.com/votingworks/vxsuite/pull/4143/commits/7b302e1969c504b43d28b1655936145b6ab3d998)

will be following up with a change to remove the `numVotes*`/`numSeats*` strings altogether, in favour of non-interpolated ones

**Pull request merged**